File: backend/core/db/enums/report_status.py
"""
Report status enumeration.
"""

from django.db import models


class ReportStatusEnum(models.TextChoices):
    """Status of a citizen report"""
    RECORDED = 'RECORDED', 'Recorded'
    IN_PROGRESS = 'IN_PROGRESS', 'In Progress'
    RESOLVED = 'RESOLVED', 'Resolved'

File: frontend/lib/features/reports/data/models/report_status.dart
/// Enumeration of report statuses.
enum ReportStatus {
  /// Enregistré
  recorded('RECORDED', 'Enregistré'),

  /// En cours
  inProgress('IN_PROGRESS', 'En cours'),

  /// Résolu
  resolved('RESOLVED', 'Résolu');

  const ReportStatus(this.value, this.label);

  /// Backend-side enum value.
  final String value;

  /// Display label.
  final String label;

  /// Creates a [ReportStatus] from the backend string value.
  static ReportStatus fromString(String value) {
    switch (value) {
      case 'RECORDED':
        return ReportStatus.recorded;
      case 'IN_PROGRESS':
        return ReportStatus.inProgress;
      case 'RESOLVED':
        return ReportStatus.resolved;
      default:
        throw ArgumentError('Unknown report status: $value');
    }
  }

  /// Converts the enum to a string for the backend.
  String toJson() => value;
}

